src/plugins/intellisense/latex-parsing.tsx

```diff
@@ -0,0 +1,211 @@
+import { MathQuillField } from "components";
+import { ExpressionAug } from "plugins/text-mode/aug/AugState";
+
+export function mapAugAST(
+  node: ExpressionAug["latex"],
+  callback: (node: ExpressionAug["latex"]) => void
+) {
+  function map(x: any) {
+    if (Array.isArray(x)) {
+      for (const child of x) {
+        map(child);
+      }
+    }
+
+    if (typeof x === "object") {
+      if (typeof x.type === "string") callback(x);
+
+      for (const [_, v] of Object.entries(x)) {
+        map(v);
+      }
+    }
+  }
+
+  map(node);
+}
+
+export function getController(mq: MathQuillField) {
+  return mq.__controller as MQController;
```

Comment:
I see this function and raise you a few times where `.__controller` is used regardless. Nicer to stick with one.

Reply:
I definitely agree here. Provides a nice barrier against `__controller` being suddenly changed. I'll make the change.

Reply:
fixed